.github/workflows/check-broken-links.yml

```diff
@@ -0,0 +1,39 @@
+name: Check Dead Links
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
this check should run on every PR as well, to verify that also new code it is not affected (also to verify that this very PR is consistent)